reference.md

```diff
@@ -0,0 +1,381 @@
+
+`deref` returns the object, or `undefined` if the object is no longer available.
+
+## WeakRef API
+
+### The `WeakRef` Constructor
+
+The `WeakRef` constructor creates a WeakRef instance for the given `target` object:
+
+```js
+ref = new WeakRef(target)
+```
+
+**Parameters**
+
+* `target` - The target object for the WeakRef instance.
+
+**Returns**
+
+* The WeakRef instance.
+
+**Notes**
+
+* `WeakRef` is a constructor function, so it throws if called as a function rather than as a constructor.
+* `WeakRef` is designed to be subclassed, and so can appear in the `extends` of a `class` definition and similar.
+
+### `WeakRef.prototype.deref`
+
+The `deref` method returns the WeakRef instance's target object, or `undefined` if the target object has been reclaimed:
+
+```js
+targetOrUndefined = weakRef.deref();
+```
+
+**Parameters**
+
+* *(none)*
+
+**Returns**
+
+* The target object of the WeakRef, or `undefined`.
+
+**Errors**
+
+* *(none)*
+
+## Notes on WeakRefs
+
+Some notes on WeakRefs:
+
+* If your code has just created a WeakRef for a target object, or has gotten a target object from a WeakRef's `deref` method, that target object will not be reclaimed until the end of the current JavaScript [job](https://tc39.es/ecma262/#job) (including any promise reaction jobs that run at the end of a script job). That is, you can only "see" an object get reclaimed between turns of the event loop. This is primarily to avoid making the behavior of any given JavaScript engine's garbage collector apparent in code&nbsp;&mdash; because if it were, people would write code relying on that behavior, which would break when the garbage collector's behavior changed. (Garbage collection is a hard problem; JavaScript engine implementers are constantly refining and improving how it works.)
+* If multiple WeakRefs have the same target, they're consistent with one another. The result of calling `deref` on one of them will match the result of calling `deref` on another of them (in the same job), you won't get the target object from one of them but `undefined` from another.
+* If the target of a WeakRef is also in a [`FinalizationRegistry`](#finalizationregistry), the WeakRef's target is cleared *before* any cleanup callback associated with the registry is called.
+* You cannot change the target of a WeakRef, it will always only ever be the original target object or `undefined` when that target has been reclaimed.
+* A WeakRef might never return `undefined` from `deref`, even if nothing strongly holds the target, because the garbage collector may never decide to reclaim the object.
+
+# `FinalizationRegistry`
+
+A finalization registry provides a way to request that a *cleanup callback* get called at some point after an object registered with the registry has been *reclaimed* (garbage collected). (Cleanup callbacks are sometimes called *finalizers*.)
+
+**NOTE:** The form of this API is under review, see [issue #187](https://github.com/tc39/proposal-weakrefs/pull/187).
```

Comment:
Probably say here, explicitly, "This document is written in terms of that PR landing."